models/product/src/product-data/types.ts

```diff
@@ -13,7 +13,7 @@ import { ValueOf } from '@commercetools-test-data/core/src/@jackfranklin/test-da
 // The base generator model. Consumers configure these fields.
 export type TProductData = Omit<ProductData, 'categories'> & {
   categories: Array<Category>;
-  searchKeyword?: Array<SearchKeyword> | null;
+  searchKeyword?: Array<SearchKeyword>;
```

Comment:
isn't it supposed to be `searchKeywords` without `?`, can't see `searchKeyword` as singular on docs 😕 
https://docs.commercetools.com/api/projects/products#productdata
```suggestion
  searchKeywords: Array<SearchKeyword>;
```

Reply:
That's a good point.
I was puzzled about having both `searchKeyword` and `searchKeywords` in the [generator](https://github.com/commercetools/test-data/blob/main/models/product/src/product-data/generator.ts#L31). Perhaps it was a mistake when they were introduced or we are missing something.
I know it was a long time ago but maybe @jaikamat can help us here.

Anyway, I don't feel comfortable renaming the field right away because, technically, that would be a breaking change.
What I've done is to add the `searchKeywords` (plural) attribute to the type.

Reply:
![image](https://github.com/user-attachments/assets/6cec7ac0-2ff4-4dbf-b76c-acd950b2df27)
 
 `searchKeyword` is used in the GraphQL transformation of this model

Reply:
>I was puzzled about having both searchKeyword and searchKeywords in the [generator](https://github.com/commercetools/test-data/blob/main/models/product/src/product-data/generator.ts#L31).

If GraphQL and REST types sufficiently diverge, we instead use the base generator to aggregate all possible fields, often with the widest type, so that consumers can pass what they need, and then use the transform target (REST or GraphQL) to pick what is needed